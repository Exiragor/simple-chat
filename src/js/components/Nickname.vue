<template>
    <div class="row">
        <div class="col-md-12">
            <div class="form-group">
                <label for="nick">Никнейм</label>
                <input v-model="nickname" id="nick" type="text" class="form-control">
                <label for="color">Цвет(hex: 6c757d)</label>
                <input v-model="nameColor" id="color" type="text" class="form-control">
            </div>
        </div>
    </div>
</template>

<script>
    export default {
        name: 'nickname',

        computed: {
            nickname: {
                get() {
                    return this.$store.getters.nickname.name
                },
                set(value) {
                    this.$store.dispatch('changeNickName', {name: value})
                },
            },
            nameColor: {
                get() {
                    return this.$store.getters.nickname.color
                },
                set(value) {
                    this.$store.dispatch('changeNickName', {color: value})
                },
            }
        }
    }
</script>