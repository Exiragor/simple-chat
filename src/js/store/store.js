import Vue from 'vue';
import Vuex from 'vuex';

Vue.use(Vuex)

const store = new Vuex.Store({
    state: {
        messages: [],
        nickname: {name: '', color: ''},
    },
    actions: {
        changeNickName({commit}, nickname) {
            commit('CHANGE_NICKNAME', nickname)
        },
        addMessage({commit}, msg) {
            commit('ADD_MESSAGE', msg)
        },
    },
    mutations: {
        CHANGE_NICKNAME(state, nickname) {
            let name = (nickname.name !== undefined) ? nickname.name : state.nickname.name
            let color = (nickname.color !== undefined) ? nickname.color : state.nickname.color
            state.nickname.name = name;
            state.nickname.color = color;
        },
        ADD_MESSAGE(state, message) {
            state.messages.push(message)
        },
    },
    getters: {
        messages(state) {
            return state.messages
        },
        nickname(state) {
            return state.nickname
        }
    }
});

export default store;
