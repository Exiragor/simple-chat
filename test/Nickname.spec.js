import { shallow } from '@vue/test-utils'
import Nickname from '@/components/Nickname.vue'
import store from '@/store/store'

describe('Nickname.vue', () => {
    it('matches snapshot', () => {
        const wrapper = shallow(Nickname, {
            store
        })

        expect(wrapper.html()).toMatchSnapshot()
    })
})