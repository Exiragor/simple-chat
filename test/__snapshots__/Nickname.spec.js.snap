// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nickname.vue matches snapshot 1`] = `
<div class="row">
  <div class="col-md-12">
    <div class="form-group">
      <label for="nick">Никнейм</label>
      <input id="nick" type="text" class="form-control">
      <label for="color">Цвет(hex: 6c757d)</label>
      <input id="color" type="text" class="form-control">
    </div>
  </div>
</div>
`;
